src/endpoints/authentication/impl.cpp

```diff
@@ -20,120 +21,484 @@
+					// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+					const auto ec = irods::base64_decode(
+						reinterpret_cast<unsigned char*>(authorization.data()),
+						authorization.size(),
+						creds.data(),
+						&size);
+					log::debug("{}: base64 - error code=[{}], decoded size=[{}]", fn, ec, size);
+
+					// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+					std::string_view sv{reinterpret_cast<char*>(creds.data()), size};
+					log::debug("{}: base64 decode credentials = [{}]", fn, sv); // TODO Don't print the password
+
+					const auto colon = sv.find(':');
+					if (colon == std::string_view::npos) {
+						return _sess_ptr->send(fail(status_type::unauthorized));
+					}
+
+					std::string username{sv.substr(0, colon)};
+					std::string password{sv.substr(colon + 1)};
+					log::debug(
+						"{}: username=[{}], password=[{}]", fn, username, password); // TODO Don't print the password
+
+					// BEGIN OG OAUTH THING
+					BodyArguments args{
+						{"client_id",
+					     irods::http::globals::oidc_configuration().at("client_id").get_ref<const std::string&>()},
+						{"grant_type", "password"},
+						{"scope", "openid"},
+						{"username", username},
+						{"password", password}};
+
+					// Query endpoint
+					nlohmann::json res_item{hit_token_endpoint(encode_body(args))};
+
+					// Assume passed, get oidc token
+					const std::string jwt_token{res_item.at("id_token").get<const std::string>()};
+
+					// Feed to JWT parser
+					auto decoded_token{jwt::decode<jwt::traits::nlohmann_json>(jwt_token)};
+
+					// Get irods username
+					// Zones?
+					// uname#zname
+					const std::string irods_name{
+						decoded_token.get_payload_json().at("irods_username").get<const std::string>()};
+
+					// Issue token?
+					static const auto seconds =
+						irods::http::globals::configuration()
+							.at(nlohmann::json::json_pointer{"/http_server/authentication/basic/timeout_in_seconds"})
+							.get<int>();
+					auto bearer_token = irods::process_stash::insert(authenticated_client_info{
+						.auth_scheme = authorization_scheme::basic,
+						.username = std::move(irods_name),
+						.expires_at = std::chrono::steady_clock::now() + std::chrono::seconds{seconds}});
+
+					response_type res_rep{status_type::ok, _req.version()};
+					res_rep.set(field_type::server, BOOST_BEAST_VERSION_STRING);
+					res_rep.set(field_type::content_type, "text/plain");
+					res_rep.keep_alive(_req.keep_alive());
+					res_rep.body() = std::move(bearer_token);
+					res_rep.prepare_payload();
+
+					return _sess_ptr->send(std::move(res_rep));
+				}
+
+				constexpr auto basic_auth_scheme_prefix_size = 6;
+				std::string authorization{iter->value().substr(pos + basic_auth_scheme_prefix_size)};
+				boost::trim(authorization);
+				log::debug("{}: Authorization value (trimmed): [{}]", fn, authorization);
+
+				constexpr auto max_creds_size = 128;
+				unsigned long size = max_creds_size;
```

Comment:
Looks like things were lost/reverted in the rebase from this point on (this line included).

Please compare the original logic for basic auth to what you have in here and restore the lines that were changed.